File: ScaleformUI_Csharp/Elements/ScaleformLiteralString.cs
namespace ScaleformUI.Elements
{
    public struct ScaleformLiteralString
    {
        private readonly string _literalString;

        public ScaleformLiteralString(string literalString)
        {
            _literalString = literalString;
        }

        public string LiteralString
        {
            get
            {
                return _literalString;
            }
        }

        public static implicit operator ScaleformLiteralString(string literalString)
        {
            return new ScaleformLiteralString(literalString);
        }

        public static bool operator ==(ScaleformLiteralString a, ScaleformLiteralString b)
        {
            return a.LiteralString == b.LiteralString;
        }

        public static bool operator !=(ScaleformLiteralString a, ScaleformLiteralString b)
        {
            return a.LiteralString != b.LiteralString;
        }

        public override bool Equals(object obj)
        {
            return obj is ScaleformLiteralString literal && this == literal;
        }

        public override int GetHashCode()
        {
            return LiteralString.GetHashCode();
        }
    }
}

File: ScaleformUI_V8/src/elements/scaleform-literal-string.ts
export class ScaleformLiteralString {
    private readonly _literalString: string;

    constructor(literalString: string) {
        this._literalString = literalString;
    }

    get LiteralString(): string {
        return this._literalString;
    }

    equals(obj: Object): boolean {
        if (!(obj instanceof ScaleformLiteralString)) {
            return false;
        }
        const otherLiteralString = obj as ScaleformLiteralString;
        return this.LiteralString === otherLiteralString.LiteralString;
    }
}

function implicitOperator(literalString: string): ScaleformLiteralString {
    return new ScaleformLiteralString(literalString);
}

function operatorEquals(a: ScaleformLiteralString, b: ScaleformLiteralString): boolean {
    return a.equals(b);
}

function operatorNotEquals(a: ScaleformLiteralString, b: ScaleformLiteralString): boolean {
    return !operatorEquals(a, b);
}